src/SubdomainChecker.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+class SubdomainChecker
+{
+    private static $initiated = false;
+    private static $isSubdomain = false;
+    private static $originalDomain = '';
+    private static $subDomain = '';
+
+    public static function init()
+    {
+        if (!self::$initiated) {
+            self::init_hooks();
+        }
+
```

Comment:
> Why did you choose to make this class static? Isn't your init() method basically a constructor? Why not put all that logic and the new CF\* classes in a constructor?
> 
> If you stick with this approach make sure to write tests for all these methods too.

We can do that but since I used a hook I wanted to go with the style I used with [ServerPush](https://github.com/cloudflare/CloudFlare-Wordpress/blob/master/src/HTTP2ServerPush.php). But if we're going to remove the logic of the error message to frontend instead of backend I can remove the `admin_notices` hook and do as you suggested.

> Is this something all plugins can use? Should we move it to cloudflare/cloudflare-plugin-backend?

Init CF\* classes in a constructor? Probably but I wouldn't add this to `cloudflare/cloudflare-plugin-backend` until we're certain this is a good solution.